src/query/api/v1/middleware/rewrite.go

```diff
@@ -255,5 +240,23 @@ func rewriteRangeInQuery(expr parser.Node, res time.Duration, multiplier int) bo
 		return nil
 	})
 
-	return updated
+	if updated {
+		return true, expr.String()
+	}
+	return false, query
+}
+
+func maybeUpdateLookback(
+	params params,
+	maxResolution time.Duration,
+	opts PrometheusRangeRewriteOptions,
+) (bool, time.Duration) {
+	resolutionBasedLookback := maxResolution * time.Duration(opts.ResolutionMultiplier) // nolint: durationcheck
+	if params.isLookbackSet && params.lookback < resolutionBasedLookback {
+		return true, resolutionBasedLookback
+	}
+	if !params.isLookbackSet && opts.DefaultLookback < resolutionBasedLookback {
+		return true, resolutionBasedLookback
+	}
+	return false, params.lookback
```

Comment:
If lookback is not set, what will this return? I think it would be better to restructure the two ifs above to:
a) determine the requested lookback (params.lookback or opts.DefaultLookback) 
b) have a single if returning `true` and `resolutionBasedLookback`, if needed. 
Will be easier to follow.